Add type annotations

North101 opened the pull request.

dmungin commented:
This is cool, I'd like to merge it. Is there any way you can resolve the merge conflicts?

North101 commented:
Hope you don't mind but made a few extra changes to simplify the data:

- side scenarios are never locked and non-side scenarios are never hidden so i've changed "locked" to "hidden" and added "side": true (code has been updated to take this into account).
- treasure data is in its own top level key in assets. Makes it act more like the treasure index.
- Added encoded scenarios v3. Simplifies the data and compresses it more.
